### PR TITLE
other: add causal masked attn

### PR DESCRIPTION
**Pull request by rebel-thkim**
# Pull Request Description

## Type of Change

- [ ] New Model Support
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Code refactoring
- [x] Other (please describe):

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have added tests that prove my fix is effective or that my feature works (If needed)

## Related Issues

----
# Conventional commit
```
type(optional scope): description
```
----
# Type candidate
  - Model Updates
    - `model`: Adding New models or Bugfix for existing models
      - ex) Add LlavaNext 
      - ex) Bugfix Whisper
  - Enhancements
    - `performance`: Optimizing some models or this library itself
      - ex) Loading RBLNModel faster
      - ex) Optimizing Memory Usage of DecoderOnlyModel
  - Code Refactor
    - `refactor`: Re-arrange class architecture, or more.
      - ex) Refactor Seq2Seq
  - Documentation
    - `doc`: Update docstring only     
  - Library Dependencies
    - `dependency`: Update requirements, something like that.
  - Other
    - `other`: None of above.
      - ex) ci update
      - ex) pdm update

### CODE REVIEW

**Review by rebel-kblee — APPROVED** [review]
review done

**Comment by rebel-jongho** [comment]
PR title을 convention에 맞추기 / pytest 통과 위한 deploy package 지정 부탁드려요

**Review by rebel-jongho — COMMENTED** [review]
pytest, seq2seq만 잘 확인해주시면 될 거 같습니다. 소스는 깔끔한 것 같은데요?

File: src/optimum/rbln/transformers/models/decoderonly/modeling_decoderonly.py [hunk 1]
```diff
@@ -110,6 +114,13 @@ def decode_forward(
...
+        if not self.use_attention_mask:
+            logits = super().forward(
+                inputs,
+                cache_position,
+            )
+            return logits
+
         if attention_mask is None:
             for b_idx in range(batch_size):
                 decoding_step = cache_position[b_idx].item()
```
[the remark this answers is not in the document]
Comment (rebel-thkim) [reply src/optimum/rbln/transformers/models/decoderonly/modeling_decoderonly.py:126]: 임시 코드라고 생각하고 대충 했던 부분이네요..ㅜ 수정하겠습니다.
